File: hw1/hw1/Calculator.cs
namespace hw1
{
    public class Calculator
    {
        public static int Calculate(Operations operation, int val1, int val2)
        {
            var result = operation switch
            {
                Operations.Plus => val1 + val2,
                Operations.Minus => val1 - val2,
                Operations.Multiply => val1 * val2,
                Operations.Divide => val1 / val2,
                _ => 0
            };
            

            return result;
        }
    }
}

File: HW5/HW5/Calculator.fs
module HW5.Calculator

open HW5.ResultBuilder

let divideBy numerator denominator =
    if denominator = decimal 0 then Error "Denominator is zero(0)"
    else Ok(numerator / denominator)
    
let calculate (val1, operation, val2) =
    result{
        match operation with
        | Operations.Plus -> return val1 + val2
        | Operations.Minus -> return val1 - val2
        | Operations.Multiply -> return val1 * val2
        | _ -> let! res =  divideBy val1 val2
               return res
    }